platform-core-module/src/main/java/com/softicar/platform/core/module/email/mailbox/imap/Office365ImapConnectorTokenCache.java

```diff
@@ -0,0 +1,49 @@
+package com.softicar.platform.core.module.email.mailbox.imap;
+
+import com.google.gson.Gson;
+import com.softicar.platform.core.module.server.AGServer;
+import java.util.Objects;
+import java.util.Optional;
+
+class Office365ImapConnectorTokenCache {
+
+	private final String username;
+	private final String accessToken;
+
+	public Office365ImapConnectorTokenCache(String username, String accessToken) {
+
+		this.username = username;
+		this.accessToken = accessToken;
+	}
+
+	public static Optional<String> getAccessToken(AGServer server) {
+
+		return Optional//
+			.ofNullable(server.getConnectorCache())
+			.map(json -> new Gson().fromJson(json, Office365ImapConnectorTokenCache.class))
+			.filter(cache -> cache.isValidFor(server.getUsername()))
+			.map(cache -> cache.getAccessToken());
+	}
+
+	public static void save(AGServer server, String accessToken) {
+
+		server//
+			.setConnectorCache(new Office365ImapConnectorTokenCache(server.getUsername(), accessToken).toJson())
```

Comment:
"connector cache" vs "token cache" - if the choice is between those two, the latter seems more appropriate to me. But how about "cached token"?

Reply:
- `cached token` does not make sense, since this could contain more than the token in the future
- renamed the field `AGServer.CONNECTOR_CACHE` to `CONNECTOR_DATA`
- renamed `Office365ImapConnectorTokenCache` to `Office365ImapConnectorData`

Reply:
please resolve this conversation if you confirm this change